spec/rubocop/cop/rspec/capybara/specific_matcher_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::Capybara::SpecificMatcher do
+  it 'does not register an offense for abstract matcher when elements with' \
+    'no matcher specified in detail' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to have_selector('article')
+      expect(page).to have_none_of_selectors('body')
+      expect(page).to have_css('tbody')
+    RUBY
+  end
+
+  it 'does not register an offense for abstract matcher when elements with' \
+    'not element' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to have_no_css('.a')
+      expect(page).to has_css?('#button')
+      expect(page).to has_no_css?('[table]')
+    RUBY
+  end
+
+  it 'registers an offense when using `#have_selector`' do
+    expect_offense(<<-RUBY)
+    expect(page).to have_selector('button')
+                    ^^^^^^^^^^^^^^^^^^^^^^^ Use `#have_button` instead of `#have_selector`.
+    expect(page).to have_selector('a')
+                    ^^^^^^^^^^^^^^^^^^ Use `#have_link` instead of `#have_selector`.
+    expect(page).to have_selector('table')
+                    ^^^^^^^^^^^^^^^^^^^^^^ Use `#have_table` instead of `#have_selector`.
+    expect(page).to have_selector('select')
+                    ^^^^^^^^^^^^^^^^^^^^^^^ Use `#have_select` instead of `#have_selector`.
+    RUBY
+  end
+
+  it 'registers an offense when using `#have_none_of_selectors`' do
+    expect_offense(<<-RUBY)
+    expect(page).to have_none_of_selectors('button')
```

Comment:
What if the `have_none_of_selectors` is used with mutiple arguments, and 'button' is not the first one?
Even if the button is the first one, `have_*_of_selectors` accept multiple locators, while other methods just one.
There is no direct substitute.

Reply:
Oh, yes, indeed. I think I'll exclude `have_*_of_selectors` as it seems to be outside the realm of this Cop.